R/transformAbundance.R

```diff
@@ -0,0 +1,400 @@
+                       call. = FALSE)
+              }
+
+              # Check name
+              if(!.is_non_empty_string(name)){
+                  stop("'name' must be a non-empty single character value.",
+                       call. = FALSE)
+              }
+
+              # Check pseudocount
+              if(!(pseudocount==TRUE || pseudocount==FALSE || pseudocount>0)){
+                  stop("'pseudocount' must be boolean or positive numeric value.",
+                       call. = FALSE)
+              }
+
+              # Check scale
+              if(!is.numeric(scale)){
+                  stop("'scale' must be a numeric value.",
+                       call. = FALSE)
+              }
+
+              # Check target
+              if(!target=="samples"){
+                  if(!transform=="Z"){
+                      stop("'target' must have value 'samples' in all but Z transform,
+                           where 'target' also can get value 'features'.",
+                           call. = FALSE)
+                  } else{
+                      if(!target=="features"){
+                          stop("'target' must be 'samples' or 'features'",
+                               call. = FALSE)
+                      }
+                  }
+              }
+
+              # Get transformed table
+              transformed_table <- .get_transformed_table(assay = assay(x, abund_values),
+                                                          transform = transform,
+                                                          pseudocount = pseudocount,
+                                                          scale = scale,
+                                                          target = target)
+
+              # Assign transformed table to assays
+              assay(x, name) <- transformed_table
+
+              return(x)
+
+          }
+)
+
+#' @rdname transformAbundance
+#' @export
+setGeneric("getTransformAbundance", signature = c("x"),
+           function(x,
+                    abund_values = "counts",
+                    transform = "identity",
+                    name = transform,
+                    pseudocount = FALSE,
+                    scale = 1,
+                    target = "samples")
+               standardGeneric("getTransformAbundance"))
+
+
+#' @rdname transformAbundance
+#' @export
+setMethod("getTransformAbundance", signature = c(x = "SummarizedExperiment"),
+          function(x,
+                   abund_values = "counts",
+                   transform = "identity",
+                   name = transform,
+                   pseudocount = FALSE,
+                   scale = 1,
+                   target = "samples"){
+
+              # Get object with transformed table
+              x <- transformAbundance(x, abund_values, transform, name, pseudocount, scale, target)
+
+              # Get the table
+              mat <- assay(x, name)
```

Comment:
writing and extracting is not necessary.

Reply:
getTransformAbundance() deleted